### PR TITLE
Refactored data driven tests

### PR DESCRIPTION
**Pull request by DavertMik**
* documentation updated
* specify dataset using `Data()`
* creating data tests moved out from `bdd.js` to keep it simple
* added `test.inject` property for not to hardcode data injection in `scenario.js`
* data driven test can accepts DataTable, array, function, generator
* renamed `scenarioContext` to `current` for better readbility

```js
// define data table inside a test or load from another module
let accounts = new DataTable(['login', 'password']); //
accounts.add(['davert', '123456']); // adding records to a table
accounts.add(['admin', '123456']);

// Pass dataTable to Data()
// Use special param `current` to get current data set
Data(accounts).Scenario('Test Login', (I, current) => {
  I.fillField('Username', current.login); // current is reserved!
  I.fillField('Password', current.password);
  I.click('Sign In');
  I.see('Welcome '+ current.login);
});
```

### CODE REVIEW

File: lib/data/context.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        data.forEach((dataRow) => {
+          opts.data = dataRow;
+          let test = context.Scenario(`${title} | ${dataRow}`, opts, fn);
+          test.inject.current = dataRow;
+        });
+      }
+    };
+  };
+
+  context.xData = function (dataTable) {
```
Comment (APshenkin) [inline lib/data/context.js]: What solution will be for Data.only?

File: test/data/sandbox/ddt_test.ddt.js [hunk 2]
```diff
@@ -1,18 +1,24 @@
+Feature('DDT');
```
Comment (APshenkin) [inline test/data/sandbox/ddt_test.ddt.js]: Previously user was be able to set datatable for Feature. This implementation doesn't allow to do this. Is this OK?

File: lib/data/context.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    return {
+      Scenario: function (title, opts, fn) {
+        if (typeof opts === 'function' && !fn) {
+          fn = opts;
+          opts = {};
+        }
+        data.forEach((dataRow) => {
+          opts.data = dataRow;
+          let test = context.Scenario(`${title} | ${dataRow}`, opts, fn);
+          test.inject.current = dataRow;
```
Comment (APshenkin) [inline lib/data/context.js]: now `current` is not able to use as pageObject name. We have to change pageObject validation here https://github.com/Codeception/CodeceptJS/blob/master/lib/container.js#L142

to block using `current` and allow `scenarioContext`

Comment (DavertMik) [inline lib/data/context.js]: It will be overridden. For non-data tests you can use `current`...

File: test/data/sandbox/ddt_test.ddt.js (hunk 2 above)
Comment (DavertMik) [inline test/data/sandbox/ddt_test.ddt.js]: You can set datatable for feature by setting the same datatable for each Scenario inside a test.
This is much more usable:

```js
Data(accounts).Scenario('do smth with users')
Data(accounts).Scenario('do smth else')
```

File: lib/data/context.js (hunk 1 above)
Comment (DavertMik) [inline lib/data/context.js]: sorry, I didn't get this

File: lib/data/context.js (hunk 3 above)
Comment (DavertMik) [inline lib/data/context.js]: Added this to documentation but removed check from container.

We need separate concerns, so container should no have hardcoded elements inside it which are related to the different parts of CodeceptJS
the only thing responsible for data management is located in `data/context`.

Comment (APshenkin) [inline lib/data/context.js]: agreed

File: test/data/sandbox/ddt_test.ddt.js (hunk 2 above)
Comment (APshenkin) [inline test/data/sandbox/ddt_test.ddt.js]: agreed

File: lib/data/context.js (hunk 1 above)
Comment (APshenkin) [inline lib/data/context.js]: Looks like it's important to have ability to launch only test that contains `Data`

Comment (DavertMik) [inline lib/data/context.js]: You mean a specific dataset? Sorry I have no ideas for now how to do it (

**Pull request merged**